### PR TITLE
feat(compliance): RHICOMPL-2566 introduce compliance templates

### PR DESCRIPTION
**Pull request by skateman**

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.redhat.cloud.notifications.processors.email.aggregators;
+
+import com.redhat.cloud.notifications.models.EmailAggregation;
+
+public class ComplianceEmailAggregator extends AbstractEmailPayloadAggregator {
+
+    public AnsibleEmailAggregator() {
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java:7]:
```suggestion
    public ComplianceEmailAggregator() {
```

File: backend/src/main/resources/templates/Compliance/reportUploadFailedEmailTitle.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+Red Hat Enterprise Linux - Compliance Instant Notification - {action.timestamp.toUtcFormat()} - Failed to uplad report of Policy '{action.content.policy}' from Host '{action.content.host}'
```
Comment (josejulio) [inline backend/src/main/resources/templates/Compliance/reportUploadFailedEmailTitle.txt:1]: typo 

```suggestion
Red Hat Enterprise Linux - Compliance Instant Notification - {action.timestamp.toUtcFormat()} - Failed to upload report of Policy '{action.content.policy}' from Host '{action.content.host}'
```

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Compliance.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    @Override
+    public boolean isEmailSubscriptionSupported(EmailSubscriptionType type) {
+        return type == EmailSubscriptionType.INSTANT;
+    }
+
+    @CheckedTemplate(requireTypeSafeExpressions = false)
+    public static class Templates {
+
+        public static native TemplateInstance complianceBelowThresholdEmailTitle();
+        public static native TemplateInstance reportUploadFailedEmailTitle();
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Compliance.java:59]:
```suggestion

        public static native TemplateInstance reportUploadFailedEmailTitle();
```

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Compliance.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        public static native TemplateInstance complianceBelowThresholdEmailBody();
+        public static native TemplateInstance reportUploadFailedEmailBody();
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Compliance.java:62]:
```suggestion

        public static native TemplateInstance reportUploadFailedEmailBody();
```

**Review by josejulio — CHANGES_REQUESTED** [review]
Could you add the following changes?
A typo and some checkstyle fixes.

Also note that the Aggregator needs to be added to the [AggregatorFactory](https://github.com/RedHatInsights/notifications-backend/blob/0a7726abb3ec1085c068a5c8b53882d00d297987/backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/EmailPayloadAggregatorFactory.java#L31-L30) but not really needed right now

**Comment by skateman** [comment]
@josejulio adjusted what you requested and finalized the HTML templates. Can you please take a look?

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**